tests/intrinsics/test_actions.py

```diff
@@ -932,3 +932,203 @@ def test_return_value_if_cannot_handle(self, can_handle_mock):
+
+    def test_second_level_key_not_string(self):
+        mappings = {
+            "MapA":{
+                "TopKey1": {
+                   "SecondKey2": "value3"
+               },
+                "TopKey2": {
+                    "SecondKey1": "value4"
+                }
+            }
+        }
+        input = {
+            "Fn::FindInMap": ["MapA", "TopKey1", ["SecondKey2"]]
+        }
+        output = self.ref.resolve_parameter_refs(input, mappings)
+
+        self.assertEqual(input, output)
+
+    def test_mapping_not_found(self):
+        mappings = {
+            "MapA":{
+                "TopKey1": {
+                   "SecondKey2": "value3"
+               },
+                "TopKey2": {
+                    "SecondKey1": "value4"
+                }
+            }
+        }
+        input = {
+            "Fn::FindInMap": ["MapB", "TopKey2", "SecondKey1"]
+        }
+        output = self.ref.resolve_parameter_refs(input, mappings)
+
+        self.assertEqual(input, output)
+
+    def test_top_level_key_not_found(self):
+        mappings = {
+            "MapA":{
+                "TopKey1": {
+                   "SecondKey2": "value3"
+               },
+                "TopKey2": {
+                    "SecondKey1": "value4"
+                }
+            }
+        }
+        input = {
+            "Fn::FindInMap": ["MapA", "TopKey3", "SecondKey1"]
+        }
+        output = self.ref.resolve_parameter_refs(input, mappings)
+
+        self.assertEqual(input, output)
+
+    def test_second_level_key_not_found(self):
+        mappings = {
+            "MapA":{
+                "TopKey1": {
+                   "SecondKey2": "value3"
+               },
+                "TopKey2": {
+                    "SecondKey1": "value4"
+                }
+            }
+        }
+        input = {
+            "Fn::FindInMap": ["MapA", "TopKey1", "SecondKey1"]
+        }
+        output = self.ref.resolve_parameter_refs(input, mappings)
+
+        self.assertEqual(input, output)
+
+    def test_one_level_find_in_mappings(self):
+        mappings = {
+            "MapA":{
+                "TopKey1": {
+                   "SecondKey2": "value3"
+               },
+                "TopKey2": {
+                    "SecondKey1": "value4"
+                }
+            }
+        }
+        input = {
+            "Fn::FindInMap": ["MapA", "TopKey2", "SecondKey1"]
+        }
+        expected = "value4"
+        output = self.ref.resolve_parameter_refs(input, mappings)
+
+        self.assertEqual(expected, output)
+
+    def test_nested_find_in_mappings(self):
+        mappings = {
+            "MapA":{
+                "TopKey1": {
+                   "SecondKey2": "value3"
+               },
+                "TopKey2": {
+                    "SecondKey1": "TopKey1"
+                }
+            }
+        }
+        input = {
+            "Fn::FindInMap": ["MapA", {"Fn::FindInMap": ["MapA", "TopKey2", "SecondKey1"]}, "SecondKey2"]
```

Comment:
Can you add a similar test, but one which looks up an entry in `MapA` based on an entry in `MapB`?